crates/transport_tx5/src/test.rs

```diff
@@ -525,9 +536,161 @@ async fn preflight_send_recv() {
     .unwrap();
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn nonexistent_peer_marked_unresponsive() {
```

Comment:
Not sure this test is necessary but it was the simpler one to write first =)

Reply:
Seems valid to me. This ties in with what you were asking about with "what if we get a peer info that's bogus and just produced to waste our time", but better put than my quick summary there :D 

I think we absolutely want to know that will be marked unavailable. 

Generally, I like simple tests that check things that seem obvious. Because just occasionally they aren't and it would have been easy to write a test for them and catch the problem